### PR TITLE
Fix seed checkpoint creation assertion

### LINKED ISSUE #1107: Creating seed checkpoint assertion
Here: https://github.com/pytorch/torchtitan/blob/2f78d7b7824aa466428ea54d5edfb44c813a1b61/torchtitan/train.py#L464C9-L467C89

Should it be == 1:
```
            assert int(
                os.environ["WORLD_SIZE"]
            ) == 1, "Must create seed checkpoint using a single device, to disable sharding."
```

### PR DESCRIPTION
**Pull request by rualark**
Fixes #1107

### CODE REVIEW

**Review by tianyu-l — COMMENTED** [review]
Thanks! Could you please fix linting?
following https://github.com/pytorch/torchtitan/blob/main/CONTRIBUTING.md#pull-requests

**Review by tianyu-l — APPROVED** [review]
LGTM! thanks!
